diff --git a/index.d.ts b/index.d.ts
index 8053bce2369b4228022d1c2132fb6bed802804e9..427c20ebdedfaf6afaf1607abb13bc7bb9fe9eda 100644
--- a/index.d.ts
+++ b/index.d.ts
@@ -172,6 +172,8 @@ declare class Config<Definitions extends Config.DefinitionsObject = Config.Defin
      * `project`, `user`, `global`, `builtin`.
      */
     save(where: Config.ConfFileType): Promise<void>;
+
+    getCredentialsByURI (uri: string): Config.Credentials;
 }
 
 declare class Umask {
@@ -253,4 +255,9 @@ declare namespace Config {
     type Problem =
         | { action: "delete"; key: string; where: ConfType }
         | { action: "rename"; from: string; to: string; where: ConfType };
+
+    type BaseCredentials = { email?: string; certfile?: string; keyfile?: string; };
+    type TokenCredentials = BaseCredentials & { token: string; };
+    type AuthCredentials = BaseCredentials & { username: string; password: string; auth: string };
+    type Credentials = TokenCredentials | AuthCredentials | BaseCredentials;
 }
diff --git a/lib/definitions/index.d.ts b/lib/definitions/index.d.ts
new file mode 100644
index 0000000000000000000000000000000000000000..29daf54ba16815b3796f673f6f46d4907c89147b
--- /dev/null
+++ b/lib/definitions/index.d.ts
@@ -0,0 +1,5 @@
+import { Options } from '../../index';
+
+export const definitions: Options['definitions'];
+export const shorthands: Options['shorthands'];
+export const flatten: NonNullable<Options['flatten']>;
